packages/core/src/odata-common/selectable/nullable-string-field.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable max-classes-per-file */
+
+import { EdmTypeShared } from '../edm-types';
+import { Entity, ODataVersionOf } from '../entity';
+import { ComplexTypeField, getEntityConstructor } from './complex-type-field';
+import { ConstructorOrField } from './constructor-or-field';
+import { EdmTypeField, SelectableEdmTypeField } from './edm-type-field';
+
+/**
+ * Represents a property with a string value, that is nullable.
+ *
+ * @typeparam EntityT - Type of the entity the field belongs to
+ */
+class NullableStringFieldBase<EntityT extends Entity> extends EdmTypeField<
+  EntityT,
+  string | null
```

Comment:
General: I know you tried a lot but it breaks my heart a bit to see so many extra classes. If I got it right the only relvant difference  is here this value for the field type which is `string` and `string|null` in the two cases right?

Reply:
Right